### PR TITLE
Support tips for public mempool submission

### PR DESCRIPTION
**Pull request by MartinquaXD**
# Description
We had issues with submitting solutions on arbitrum whenever the gas prices went above the 0.01 Gwei it usually is. To address that we configured an additional tip. That worked well while we submitted via `merkle`. Since we stopped using `merkle` we see a bunch of [errors](https://aws-es.cow.fi/_dashboards/app/data-explorer/discover#?_a=(discover:(columns:!(log,log_level),isDirty:!f,sort:!()),metadata:(indexPattern:'86e4a5a0-4e4b-11ef-85c5-3946a99ed1a7',view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2024-09-08T19:45:08.598Z',to:'2024-10-08T01:21:43.919Z'))&_q=(filters:!(),query:(language:kuery,query:'%22gas:%20max%20fee%20per%20gas%20less%20than%20block%20base%20fee%22%20AND%20%22arbitrum%22'))) indicating that we provide a too low gas price for the transactions.
Turns out we only have the ability to configure additional tips using private mempools but not with the public mempool. That's way we stopped seeing those errors while `merkle` was active.

# Changes
Added the ability to also configure gas price tips when submitting via public mempool.

Will create the necessary follow up PR afterwards.

## How to test
Compiler

### CODE REVIEW

File: crates/driver/src/infra/blockchain/gas.rs [hunk 1]
```diff
@@ -39,13 +39,17 @@ impl GasPriceEstimator {
...
                     max_additional_tip,
                     additional_tip_percentage,
                     ..
-                } => Some((max_additional_tip, additional_tip_percentage)),
-                mempool::Kind::Public(_) => None,
+                } => (max_additional_tip, additional_tip_percentage),
+                mempool::Kind::Public {
+                    max_additional_tip,
+                    additional_tip_percentage,
+                    ..
```
Comment (squadgazzz) [inline crates/driver/src/infra/blockchain/gas.rs:51]: Are these dots required here?
```suggestion
```

File: crates/driver/src/infra/config/file/load.rs [hunk 2]
```diff
@@ -264,22 +264,30 @@ pub async fn load(chain: eth::ChainId, path: &Path) -> infra::Config {
                 target_confirm_time: config.submission.target_confirm_time,
                 retry_interval: config.submission.retry_interval,
                 kind: match mempool {
-                    file::Mempool::Public => {
+                    file::Mempool::Public {
+                        max_additional_tip,
+                        additional_tip_percentage,
+                        ..
```
Comment (squadgazzz) [inline crates/driver/src/infra/config/file/load.rs:270]: Are these dots required here?
```suggestion
```

**Review by squadgazzz — APPROVED** [review]
LGTM

File: crates/driver/src/infra/config/file/load.rs (hunk 2 above)
Comment (MartinquaXD) [reply crates/driver/src/infra/config/file/load.rs]: Nope, not needed. Removed.

**Pull request merged**